### ISSUE TITLE
Slow Training Time in GRPO

### ISSUE THREAD

**Issue by troy12x** [huggingface/open-r1#161]
I have 8xA100 80gb sxm4 gpu's and I'm using the same config as you guys I didn’t change anything. But still, my training log shows:

{'loss': -0.0, 'grad_norm': 0.1909245824751648, 'learning_rate': 2.0000000000000002e-07, 'completion_length': 827.9754867553711, 'rewards/accuracy_reward': 0.02566964435391128, 'rewards/format_reward': 0.011160714901052415, 'reward': 0.03683035890571773, 'reward_std': 0.09278179239481688, 'kl': 0.0, 'epoch': 0.0}
  0%|                                                                              | 1/1000 [04:41<**78:02:42, 281.24s/it]**

I’m getting a training time of 78 hours just for GRPO! Why is that? I'm using the latest everything is this normal? even with vllm?

**Comment by StarDewXXX** [comment]
It seems that Open-r1 only use one GPU for vllm generation, but generation is the most time-cost part of GRPO training

**Comment by GaryZhu1996** [comment]
Hello, I hope you're doing well. @troy12x 
I would like to ask about the model you used for training and the dataset you selected. How is the performance of your trained model?
I have been working followed the Open-R1 framework to use the NuminaMath-TIR dataset, but the models I trained based on the native Qwen2.5-7B or Deeseek-Distill-Qwen7B have not yielded satisfactory results. If possible, could you kindly share some insights into your parameter settings? I would greatly appreciate your guidance.
